website/models/user/user_tag.py

```diff
@@ -14,4 +15,12 @@ class UserTagRelationship(db.Model):
     __tablename__ = 'user_tag_relationships'
     id: Mapped[int] = mapped_column(Integer, primary_key=True)
     user_id: Mapped[int] = mapped_column(Integer, ForeignKey('users.id'))
-    tag_id: Mapped[int] = mapped_column(Integer, ForeignKey('user_tags.id'))
\ No newline at end of file
+    tag_id: Mapped[int] = mapped_column(Integer, ForeignKey('user_tags.id'))
+
+class Roles(enum.Enum):
+    Choreographer: str = "Choreographer"
+    Designer: str = "Designer"
+    Writer: str = "Writer"
+    Producer: str = "Producer"
+    StageManager: str = "Stage Manager"
+    Other: str = "Other"
```

Comment:
I'm not sure I understand why we have these Roles. Didn't we go with tags so that we wouldn't be prescriptive in what people could display themselves as? Perhaps we could offer these roles as preset tags the user can select in addition to the free-entry text box?

Reply:
These weren't really meant for limiting the possible tags to be stored database, I just wanted to make a list of role options for the filter on the people search page to make it easier to load each option. Probably should place this in `forms/people_filter.py`.

But I'll add a free-entry text box underneath the `Other` option, so they can specify specific tags like what you added for the events filter page.

Maybe later on, we could add a `+` button next to the filter which would pop up a new search bar allowing the user to add a new tag by search existing tags or just typing in their own tag. Similarly, to how you can add/delete new event occurrences in the create event form. Then, once the tag is added to the filter it would appear like it would in the user card, but with an `x` button to remove it from the filter. But it might be complicated to implement so we can leave it as free-text for now.

Something like this (apologies for Paint graphics):
![image](https://github.com/AndrewHocking/DanceConnect/assets/77180065/3189c6d0-dd58-4e7e-a847-e2a78fe145ba)

Reply:
Oh awesome, this makes a lot of sense! I like this a lot

Reply:
I added the free-text filter for now, let me know if it's okay. Maybe we can make updating the tag filter UI as a low priority issue.